### DISCUSSION TITLE
Copy text, copy image to clipboard

### DISCUSSION THREAD

**Discussion by JunglePredator** [pdfarranger/pdfarranger#905] [Ideas]
I don't know if it's what you intend for the scope of the project but integrating some of poppler's features to pdf arrange would make it a better all in one solution.

Like right now you can only extract a page to another pdf; with poppler integration it'd allow users to extract to text or the images on a page etc.

It of course would add quite a bit of work for things users can already do; it'd make the work flow a lot nicer however.

Thought it'd be nice to see what other users think.

**Comment by dreua** [comment]
Poppler is already integrated, it renders the previews. Can you edit the title to be more specific?
However, what you are suggesting is most likely out of scope. I have two thoughts anyway:
- Does Poppler even provide the features you are describing? (Maybe add links to example code.)
- Libreoffice Draw can open PDFs, does it provide the features you are missing? Should we maybe mention it as an alternative for this use case?

**Comment by jeromerobert** [comment]
https://gitlab.freedesktop.org/poppler/poppler/-/blob/master/utils/pdftotext.cc
https://gitlab.freedesktop.org/poppler/poppler/-/blob/master/utils/pdfimages.cc

First step is to check if the required API exists in the Python API.

**Comment by kbengs** [comment]
A idea I was thinking about when working with overlays & underlays was to have a "explode" action that would explode a page into its pieces. These pieces would then be added to iconview as new pages. Similar could probably be done with pages with images. A image could then be copied to clipboard with a "copy image to clipboard" action. Pikepdf support extraction of images but not text.

**Comment by JunglePredator** [comment]
Yes, there's already multiple programs that can do the various options; that's why I put this as a discussion.
Jerome linked the appropriate code for what I was specifically mentioning.
I thought to bring it up just to see if pdf arranger would make a better 'complete' solution without massive changes. It's just a work flow idea so one doesn't have to save-open-save-open et cetera.
For kbengs's idea... there's already a freeware pdf program that can basically sperate pages into its component pieces... but I can't remember which one it is. :-/

**Comment by jeromerobert** [comment]
https://lazka.github.io/pgi-docs/Poppler-0.18/classes/Page.html#Poppler.Page.get_text
https://lazka.github.io/pgi-docs/Poppler-0.18/classes/Page.html#Poppler.Page.get_image_mapping
https://lazka.github.io/pgi-docs/Poppler-0.18/classes/Page.html#Poppler.Page.get_image
https://lazka.github.io/pgi-docs/index.html#Gdk-4.0/functions.html#Gdk.pixbuf_get_from_surface

**Comment by jeromerobert** [comment]
Moved to #927
